### ISSUE TITLE
Trailing comma incorrectly identified inside multiline arrow function

### ISSUE THREAD

**Issue by borekb** [palantir/tslint#2795]
### Bug Report

- __TSLint version__: 5.2.0
- __TypeScript version__: 2.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
commitsTableStore.setCommitRows(data.commits.map(commit =>
  new CommitRow(commit) // missing trailing comma reported here
));
```

with `tslint.json` configuration:

```json
"trailing-comma": [true, {"multiline": "always"}]
```

#### Actual behavior

Missing trailing comma reported.

#### Expected behavior

No linting error reported.

#### More info

It's caused by the "functions" checker – TSLint thinks that this is a multiline function call.

**Comment by adidahiya** [comment]
huh, you're right, that is a false positive. accepting bugfix PRs

**Comment by ajafff** [comment]
Cross posting my comment from https://github.com/palantir/tslint/pull/2988#issuecomment-312840722:

IMO this is not a bug or false positive. 

The above example is considered "multiline" for a reason. The docs are very clear on that:

> An array is considered “multiline” if its closing bracket is on a line after the last array element. The same general logic is followed for object literals, function typings, named import statements and function parameters.

To avoid this, you can simply rewrite it as follows:

```ts
commitsTableStore.setCommitRows(data.commits.map(commit =>
  new CommitRow(commit)));

//or
commitsTableStore.setCommitRows(data.commits.map((commit =>
  new CommitRow(commit)
))); //  note the extra parens around the arrow function
```

If we want to implement an exception for such cases, this would need a proposal what exacltly should be handled different. And it should be implemented behind an option.

**Comment by adidahiya** [comment]
ah, I see your point @ajafff, and I now agree, this is not a bug or false positive. as per https://github.com/palantir/tslint/issues/2909#issuecomment-307898234, we don't have plans to to check the arity of the function being called in the trailing-comma implementation. your proposed workarounds are good 👍

**Comment by borekb** [comment]
Thanks for all the context. I have to say it's quite unintuitive at first but follows some internal logic. And I didn't know you can put trailing commas in function calls just like that :)

For reference, this is how Prettier handles the situation with `--trailing-comma=es5` configuration: https://goo.gl/Juay94

```
commitsTableStore.setCommitRows(
  data.commits.map(
    commit => new CommitRow(commit) // missing trailing comma reported here
  )
);
```
